src/Serilog/project.json

```diff
@@ -31,7 +19,7 @@
         "dotnet5.1": {
             "compilationOptions": {
                 "keyFile": "../../assets/Serilog.snk",
-                "define": [ "NO_APPDOMAIN" ]
+                "define": [ "ASYNCLOCAL", "NO_APPDOMAIN" ]
```

Comment:
No, unfortunately it is not. It shouldn't have the `ASYNCLOCAL` either, but that was harmless because of the missing `LOGCONTEXT`. Understanding https://github.com/dotnet/corefx/blob/c747d929093681a475559f804043cb5ac5470a49/Documentation/project-docs/standard-platform.md is apparently completely beyond me, because my interpretation is that `dotnet5.1` should run on _.Net Framework 4.5_, but attempting to compile that way results in errors about CallContext not being available (it came with 4.5 AFAIK), `dotnet5.2` which I understood to mean _.Net Framework 4.5.1_ and up results in even more errors...

I will remove the ASYNCLOCAL and I cleaned up another bit of #if !NET45 nonsense, although I've still got a bit #if !DOTNET5_1 around the entire `EnvironmentUserNameEnricher`